components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,90 +486,360 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
-                            filterOperation));
-                }
+            List<Object> filteredUsers = new ArrayList<>();
 
-                if (SCIMCommonUtils.isFilteringEnhancementsEnabled()) {
-                    if (SCIMCommonConstants.EQ.equalsIgnoreCase(filterOperation)) {
-                        if (StringUtils.equals(attributeName, SCIMConstants.UserSchemaConstants.USER_NAME_URI) &&
-                                !StringUtils.contains(attributeValue, CarbonConstants.DOMAIN_SEPARATOR)) {
-                            attributeValue = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME +
-                                    CarbonConstants.DOMAIN_SEPARATOR + attributeValue;
-                        }
-                    }
+            // 0th index is to store total number of results.
+            filteredUsers.add(0);
+
+            // Support multi attribute filtering.
+            return getMultiAttributeFilteredUsers(node, requiredAttributes, offset, limit, sortBy, sortOrder,
+                    domainName, filteredUsers);
+        } else {
+            throw new CharonException("Unknown operation. Not either an expression node or an operation node.");
+        }
+    }
+
+    /**
+     * Method to filter users for a filter with a single attribute.
+     *
+     * @param node               Expression node for single attribute filtering
+     * @param requiredAttributes Required attributes for the response
+     * @param offset             Starting index of the count
+     * @param limit              Counting value
+     * @param sortBy             SortBy
+     * @param sortOrder          Sorting order
+     * @param domainName         Domain to run the filter
+     * @return User list with detailed attributes
+     * @throws CharonException Error while filtering
+     */
+    private List<Object> getFilteredUsersBySingleAttribute(ExpressionNode node, Map<String, Boolean> requiredAttributes,
+            int offset, int limit, String sortBy, String sortOrder, String domainName) throws CharonException {
+
+        String[] userNames;
+
+        // Set filter values.
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    String.format("Listing users by filter: %s %s %s", attributeName, filterOperation, attributeValue));
+        }
+        try {
+
+            validateFilterOperation(filterOperation);
+
+            // Apply filter enhancements for username and update expression node value.
+            attributeValue = applyFilterEnhancementsForUsername(filterOperation, attributeName, attributeValue, node);
+
+            if ((carbonUM instanceof PaginatedUserStoreManager) && isPaginatedUserStoreAvailable()) {
+
+                // Set the limit if specified.
+                if (limit <= 0)
+                    limit = getMaxLimit();
```

Comment:
Isn't the case was that this limit is not previously applied for single attribute filtering and getting applied now.
What would happen for someone who had been using single attribute filtering moved to the version with the fix

Reply:
With new API, however, we need to pass the limit, in order to make backwards compatibility that is, if the count is not specified we need to return all the users.  We can pass Max integer value then we'll able to get all the users.

Reply:
Fixed. d7a9643d75bb2840075877c74075ff640197ed50